src/uct/ib/rdmacm/rdmacm_iface.c

```diff
@@ -70,6 +71,230 @@ static uct_iface_ops_t uct_rdmacm_iface_ops = {
     .iface_get_address        = uct_rdmacm_iface_get_address
 };
 
+void uct_rdmacm_iface_client_start_next_ep(uct_rdmacm_iface_t *iface)
+{
+    size_t ip_len = ucs_max(INET_ADDRSTRLEN, INET6_ADDRSTRLEN);
+    char *ip_str  = ucs_alloca(ip_len);
+    unsigned set_new_ep = 0;
+
+    UCS_ASYNC_BLOCK(iface->super.worker->async);
+    /* set a new ep (take one from the pending eps queue) */
+    while (!ucs_list_is_empty(&iface->pending_eps_list) && (!set_new_ep)) {
```

Comment:
instead of using 'set_new_ep', can exit the look by 'break'. also, iface->ep can be set to NULL in the beginning of the function